File: service-trade-marshals/src/main/java/com/fidelity/clientportfolio/Trade.java
package com.fidelity.clientportfolio;

public class Trade {
	private String instrumentId;
    private int quantity;
    private double executionPrice;
    private String direction;
    private String clientId;
    private Order order;
    private String tradeId;
    private double cashValue;

    // Constructor
    public Trade(String instrumentId, int quantity, double executionPrice, String direction, String clientId,
                 Order order, String tradeId, double cashValue) {
        this.instrumentId = instrumentId;
        this.quantity = quantity;
        this.executionPrice = executionPrice;
        this.direction = direction;
        this.clientId = clientId;
        this.order = order;
        this.tradeId = tradeId;
        this.cashValue = cashValue;
    }


    // Getters and Setters
    public String getInstrumentId() {
        return instrumentId;
    }

    public void setInstrumentId(String instrumentId) {
        this.instrumentId = instrumentId;
    }

    public int getQuantity() {
        return quantity;
    }

    public void setQuantity(int quantity) {
        this.quantity = quantity;
    }

    public double getExecutionPrice() {
        return executionPrice;
    }

    public void setExecutionPrice(double executionPrice) {
        this.executionPrice = executionPrice;
    }

    public String getDirection() {
        return direction;
    }

    public void setDirection(String direction) {
        this.direction = direction;
    }

    public String getClientId() {
        return clientId;
    }

    public void setClientId(String clientId) {
        this.clientId = clientId;
    }

    public Order getOrder() {
        return order;
    }

    public void setOrder(Order order) {
        this.order = order;
    }

    public String getTradeId() {
        return tradeId;
    }

    public void setTradeId(String tradeId) {
        this.tradeId = tradeId;
    }

    public double getCashValue() {
        return cashValue;
    }

    public void setCashValue(double cashValue) {
        this.cashValue = cashValue;
    }

}

File: frontend-trade-marshals/src/app/models/Trade/trade.ts
import { Order } from "./order";

export class Trade {
    constructor(
        public instrumentId: string,
        public quantity: number,
        public executionPrice: number,
        public direction: string,
        public clientId: string,
        public order: Partial<Order>,
        public tradeId: string,
        public cashValue: number,
        public executedAt: Date
    ) {}
}

// {
// 	"instrumentId": "N123456",
// 	"quantity": 10,
// 	"executionPrice": 104.25,
// 	"direction": "S",
// 	"clientId": "123",
// 	"order": {
// 		"instrumentId": "N123456",
// 		"quantity": 10,
// 		"targetPrice": 104,
// 		"direction": "S",
// 		"clientId": "123",
// 		"token": 920141621
// 	},
// 	"tradeId": "hgmdut8fdkl-9mc1zs9t54k-pm12o2amhzj",
// 	"cashValue": 1052.925
// }


